datafaker-gen-examples/datafaker-gen-bigquery/src/main/java/net/datafaker/datafaker_gen/sink/BigQueryConnectionConfiguration.java

```diff
@@ -0,0 +1,96 @@
+package net.datafaker.datafaker_gen.sink;
+
+import com.google.auth.Credentials;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.auth.oauth2.ServiceAccountCredentials;
+import com.google.cloud.bigquery.storage.v1.TableName;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Map;
+
+public class BigQueryConnectionConfiguration implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+    private final Credentials credentials;
+
+    private final String project;
+    private final String dataset;
+    private final String table;
+    private final long maxOutstandingElementsCount;
+    private final long maxOutstandingRequestBytes;
+
+    private final boolean createIfNotExists;
+
+    public BigQueryConnectionConfiguration(
+            String project,
+            String dataset,
+            String table,
+            boolean createIfNotExists,
+            long maxOutstandingElementsCount,
+            long maxOutstandingRequestBytes,
+            Credentials credentials
+    ) {
+        this.project = project;
+        this.dataset = dataset;
+        this.table = table;
+        this.createIfNotExists = createIfNotExists;
+        this.credentials = credentials;
+        this.maxOutstandingElementsCount = maxOutstandingElementsCount;
+        this.maxOutstandingRequestBytes = maxOutstandingRequestBytes;
+    }
+
+    public static BigQueryConnectionConfiguration fromConfig(Map<String, ?> config) {
+        final Credentials credentials;
+
+        String serviceAccount = (String) config.get("service_account");
+        try (FileInputStream fis = new FileInputStream(serviceAccount)) {
+            if (config.get("test") != null) {
```

Comment:
`test` why so strange name?

Reply:
I've added an additional property responsible for the type of credential loader. Now, we can support the following approaches for loading credentials:
  - **service_account**: This starts with ServiceAccountCredential. - This is Default loader.
  - **application_default**: This is the default approach from Google. This means that credential loader following chain: ComputeCredential (if not available, then) -> UserCredential (if not available, then) -> ServiceAccountCredential.
  - **Test**: This simply creates a dummy OAuth2Credentials.